diff --git a/llvm/cmake/modules/AddLLVM.cmake b/llvm/cmake/modules/AddLLVM.cmake
index 5e9896185528..eea19456b3a0 100644
--- a/llvm/cmake/modules/AddLLVM.cmake
+++ b/llvm/cmake/modules/AddLLVM.cmake
@@ -901,6 +901,20 @@ macro(add_llvm_library name)
   else()
     set_target_properties(${name} PROPERTIES FOLDER "Libraries")
   endif()
+  get_target_property(type ${name} TYPE)
+
+  if(${type} STREQUAL "STATIC_LIBRARY")
+    # Install any PDBs generated for this library
+    set(pdb_dir ${CMAKE_CURRENT_BINARY_DIR}/pdb)
+    set_target_properties(${name}
+      PROPERTIES
+      COMPILE_PDB_OUTPUT_DIRECTORY ${pdb_dir}
+    )
+    install(DIRECTORY ${pdb_dir}
+      DESTINATION lib${LLVM_LIBDIR_SUFFIX}
+      OPTIONAL
+    )
+  endif()
 endmacro(add_llvm_library name)
 
 macro(generate_llvm_objects name)
