diff --git a/cmake/Modules/CMakePolicy.cmake b/cmake/Modules/CMakePolicy.cmake
index 0ec32ad8637f..0aa7ed414de0 100644
--- a/cmake/Modules/CMakePolicy.cmake
+++ b/cmake/Modules/CMakePolicy.cmake
@@ -10,3 +10,9 @@ endif()
 if(POLICY CMP0116)
   cmake_policy(SET CMP0116 OLD)
 endif()
+
+# Enable debug info compatible with SCCache
+# https://github.com/mozilla/sccache?tab=readme-ov-file#usage
+if(POLICY CMP0141)
+  cmake_policy(SET CMP0141 NEW)
+endif()
